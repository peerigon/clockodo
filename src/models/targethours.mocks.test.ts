import { describe, expect, test } from "vitest";
import {
  createTargethoursRowMocks,
  createTargethoursRowMonthlyMocks,
  createTargethoursRowWeeklyMocks,
} from "./targethours.mocks.js";

describe("createTargethoursRowMocks()", () => {
  test("It creates realistic mocks", () => {
    expect(createTargethoursRowMocks({ count: 20 })).toMatchSnapshot();
  });

  test("Creates targethours rows on a single day", () => {
    for (let i = 0; i < 20; i++) {
      const [targethoursRow] = createTargethoursRowMonthlyMocks({
        count: 1,
        dateSinceBetween: [new Date(2020, 0, 1), new Date(2020, 0, 1)],
      });

      expect(targethoursRow).toMatchObject({
        dateSince: "2020-01-01",
      });
    }
  });
  for (let i = 0; i < 20; i++) {
    const [targethoursRow] = createTargethoursRowWeeklyMocks({
      count: 1,
      dateSinceBetween: [new Date(2020, 0, 1), new Date(2020, 0, 1)],
    });

    expect(targethoursRow).toMatchObject({
      dateSince: "2020-01-01",
    });
    expect(["2020-01-01", null]).toContain(targethoursRow.dateUntil);
  }
});
