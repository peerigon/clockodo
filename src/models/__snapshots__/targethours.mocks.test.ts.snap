// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTargethoursRowMocks() It creates realistic mocks 1`] = `
Array [
  Object {
    "compensationMonthly": 6,
    "dateSince": "2019-03-01",
    "dateUntil": "2019-04-30",
    "id": 0,
    "monthlyTarget": 48093,
    "testData": false,
    "type": "monthly",
    "usersId": 0,
    "workdayFriday": true,
    "workdayMonday": false,
    "workdaySaturday": true,
    "workdaySunday": false,
    "workdayThursday": false,
    "workdayTuesday": false,
    "workdayWednesday": false,
  },
  Object {
    "compensationMonthly": 3,
    "dateSince": "2019-06-01",
    "dateUntil": "2019-07-31",
    "id": 1,
    "monthlyTarget": 44025,
    "testData": false,
    "type": "monthly",
    "usersId": 0,
    "workdayFriday": true,
    "workdayMonday": false,
    "workdaySaturday": false,
    "workdaySunday": true,
    "workdayThursday": true,
    "workdayTuesday": false,
    "workdayWednesday": true,
  },
  Object {
    "compensationMonthly": 2,
    "dateSince": "2019-09-01",
    "dateUntil": "2019-09-30",
    "id": 2,
    "monthlyTarget": 65472,
    "testData": false,
    "type": "monthly",
    "usersId": 0,
    "workdayFriday": false,
    "workdayMonday": false,
    "workdaySaturday": true,
    "workdaySunday": true,
    "workdayThursday": false,
    "workdayTuesday": false,
    "workdayWednesday": false,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 48,
    "compensationMonthly": 4,
    "dateSince": "2020-01-13",
    "dateUntil": "2020-02-18",
    "friday": 5,
    "id": 3,
    "monday": 0,
    "saturday": 4,
    "sunday": 0,
    "testData": false,
    "thursday": 4.16,
    "tuesday": 24,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 53,
    "compensationMonthly": 2,
    "dateSince": "2020-02-19",
    "dateUntil": "2020-03-13",
    "friday": 0,
    "id": 4,
    "monday": 24,
    "saturday": 0,
    "sunday": 24,
    "testData": false,
    "thursday": 2.74,
    "tuesday": 3,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 6,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 32,
    "compensationMonthly": 2,
    "dateSince": "2020-03-18",
    "dateUntil": "2020-04-15",
    "friday": 3,
    "id": 5,
    "monday": 5.36,
    "saturday": 0,
    "sunday": 5.36,
    "testData": false,
    "thursday": 6,
    "tuesday": 0,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 24,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 12,
    "compensationMonthly": 1,
    "dateSince": "2020-04-30",
    "dateUntil": "2020-06-26",
    "friday": 0,
    "id": 6,
    "monday": 6.11,
    "saturday": 24,
    "sunday": 6.11,
    "testData": false,
    "thursday": 0,
    "tuesday": 1,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 5,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 18,
    "compensationMonthly": 4,
    "dateSince": "2020-06-27",
    "dateUntil": "2020-08-10",
    "friday": 4,
    "id": 7,
    "monday": 0,
    "saturday": 24,
    "sunday": 0,
    "testData": false,
    "thursday": 5.02,
    "tuesday": 4,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 55,
    "compensationMonthly": 4,
    "dateSince": "2020-08-11",
    "dateUntil": "2020-10-25",
    "friday": 5,
    "id": 8,
    "monday": 1,
    "saturday": 1.22,
    "sunday": 1,
    "testData": false,
    "thursday": 0,
    "tuesday": 0,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 24,
  },
  Object {
    "absenceFixedCredit": false,
    "compensationDaily": 14,
    "compensationMonthly": 1,
    "dateSince": "2020-10-26",
    "dateUntil": "2020-11-03",
    "friday": 7.41,
    "id": 9,
    "monday": 5,
    "saturday": 24,
    "sunday": 5,
    "testData": false,
    "thursday": 2,
    "tuesday": 0,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 3,
    "compensationMonthly": 3,
    "dateSince": "2020-11-04",
    "dateUntil": "2020-11-08",
    "friday": 5.64,
    "id": 10,
    "monday": 24,
    "saturday": 6,
    "sunday": 24,
    "testData": false,
    "thursday": 0,
    "tuesday": 2,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0,
  },
  Object {
    "absenceFixedCredit": false,
    "compensationDaily": 53,
    "compensationMonthly": 0,
    "dateSince": "2020-11-09",
    "dateUntil": "2020-12-14",
    "friday": 24,
    "id": 11,
    "monday": 1.21,
    "saturday": 0,
    "sunday": 1.21,
    "testData": false,
    "thursday": 8,
    "tuesday": 3,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 11,
    "compensationMonthly": 0,
    "dateSince": "2020-12-15",
    "dateUntil": "2020-12-25",
    "friday": 0,
    "id": 12,
    "monday": 24,
    "saturday": 0,
    "sunday": 24,
    "testData": false,
    "thursday": 6,
    "tuesday": 3,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0.84,
  },
  Object {
    "absenceFixedCredit": false,
    "compensationDaily": 52,
    "compensationMonthly": 2,
    "dateSince": "2020-12-28",
    "dateUntil": "2021-01-05",
    "friday": 0,
    "id": 13,
    "monday": 24,
    "saturday": 8,
    "sunday": 24,
    "testData": false,
    "thursday": 3,
    "tuesday": 6.84,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 18,
    "compensationMonthly": 3,
    "dateSince": "2021-01-06",
    "dateUntil": "2021-03-29",
    "friday": 24,
    "id": 14,
    "monday": 0,
    "saturday": 4,
    "sunday": 0,
    "testData": false,
    "thursday": 2.71,
    "tuesday": 4,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0,
  },
  Object {
    "absenceFixedCredit": true,
    "compensationDaily": 17,
    "compensationMonthly": 8,
    "dateSince": "2021-03-30",
    "dateUntil": "2021-05-19",
    "friday": 0,
    "id": 15,
    "monday": 7.25,
    "saturday": 4,
    "sunday": 7.25,
    "testData": false,
    "thursday": 24,
    "tuesday": 7,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 0,
  },
  Object {
    "absenceFixedCredit": false,
    "compensationDaily": 38,
    "compensationMonthly": 2,
    "dateSince": "2021-06-30",
    "dateUntil": "2021-07-12",
    "friday": 4,
    "id": 16,
    "monday": 0,
    "saturday": 2.06,
    "sunday": 0,
    "testData": false,
    "thursday": 24,
    "tuesday": 0,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 6,
  },
  Object {
    "absenceFixedCredit": false,
    "compensationDaily": 3,
    "compensationMonthly": 7,
    "dateSince": "2021-07-20",
    "dateUntil": "2021-09-24",
    "friday": 4.81,
    "id": 17,
    "monday": 0,
    "saturday": 24,
    "sunday": 0,
    "testData": false,
    "thursday": 0,
    "tuesday": 4,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 1,
  },
  Object {
    "absenceFixedCredit": false,
    "compensationDaily": 49,
    "compensationMonthly": 0,
    "dateSince": "2021-09-25",
    "dateUntil": "2021-10-14",
    "friday": 1,
    "id": 18,
    "monday": 0,
    "saturday": 24,
    "sunday": 0,
    "testData": false,
    "thursday": 0,
    "tuesday": 3.6,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 8,
  },
  Object {
    "absenceFixedCredit": false,
    "compensationDaily": 43,
    "compensationMonthly": 2,
    "dateSince": "2021-10-31",
    "dateUntil": "2021-11-20",
    "friday": 4,
    "id": 19,
    "monday": 24,
    "saturday": 0,
    "sunday": 24,
    "testData": false,
    "thursday": 0,
    "tuesday": 3.66,
    "type": "weekly",
    "usersId": 0,
    "wednesday": 3,
  },
]
`;
